web/app/scripts/src/directives/proportion-field.es.js

```diff
@@ -10,12 +10,48 @@ const module = angular.module(moduleProperties.moduleName, [
     ProportionField.moduleName
 ]);
 
+module.directive('ensureInteger', [
+    function () {
+        return {
+            require: 'ngModel',
+            restrict: 'A',
+            link: function (scope, element, attr, ctrl) {
+                ctrl.$parsers.push((val) => {
+                    if (val) {
+                        const value = parseInt(val, 10);
+                        if (Number.isNaN(value)) {
+                            return undefined;
+                        }
+                        return value;
+                    }
+                    return null;
+                });
+            }
+        };
+    }
+]);
+
+module.directive('integerField', [
```

Comment:
This directive seems a bit overkill for me, you might as well stick in the contents of the template straight into `proportion-field.html`